google/cloud/spanner_dbapi/parse_utils.py

```diff
@@ -509,11 +518,37 @@ def sql_pyformat_args_to_spanner(sql, params):
             resolved_value = pyfmt % params
             named_args[key] = resolved_value
         else:
+            assert_numeric_precision_and_scale(params[i])
             named_args[key] = params[i]
 
     return sanitize_literals_for_upload(sql), named_args
 
 
+def assert_numeric_precision_and_scale(value):
```

Comment:
I think this logic would make more sense to be added to the base client [here](https://github.com/googleapis/python-spanner/blob/master/google/cloud/spanner_v1/_helpers.py#L131-L132).

Reply:
moved it to base client.